### PR TITLE
feat: new batch service

### PR DESCRIPTION
**Pull request by Ubisoft-potato**
### Summary
- Currently we have 6 batch jobs running in 3 batch server. 
  Reference Issue: https://github.com/bucketeer-io/bucketeer/issues/428
- This PR will create a new batch service that collects these batch jobs together and exposes a gRPC API to be called by a Kubernetes cron job.
- The Kubernetes CronJob will be created in another pull request.

### Implementation
~~Due to all batch jobs implement the `job.Job` interface. So we can utilize the proxy design pattern to delegate the original jobs, allowing for seamless migration of these batch jobs without the need to copy or move the original batch job code file.~~

We are planning to move the original batch jobs code from the following 3 main files:
* pkg/experiment/cmd/batch/batch.go
* pkg/notification/sender/informer/batch/job.go
* pkg/opsevent/cmd/batch/batch.go

So this PR will implement these batch jobs again, and once the new batch service working as exceptation I will delete these jobs from original pacakage above.

### CODE REVIEW

**Comment by masaaania** [comment]
Great job 👏 👏 
Thank you, LGTM!

**Comment by Ubisoft-potato** [comment]
> Great job 👏 👏 Thank you, LGTM! I would check the manifest files as well and approve them.

No problem!

File: pkg/job/job.go
[the remark this answers is not in the document]
Comment (Ubisoft-potato) [reply pkg/job/job.go:1]: It make sense. 
I will move it to the `batch` package. After this PR been merged I'll delete `job` package in a new PR.

[the remark this answers is not in the document]
Comment (Ubisoft-potato) [reply pkg/job/job.go:1]: Done!

File: pkg/batch/cmd/server/server.go [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+	go healthChecker.Run(ctx)
+
+	server := rpc.NewServer(service, *s.certPath, *s.keyPath,
+		"batch-server",
+		rpc.WithPort(*s.port),
+		rpc.WithMetrics(registerer),
+		rpc.WithLogger(logger),
+		rpc.WithHandler("/health", healthChecker),
+	)
+	defer server.Stop(10 * time.Second)
```
[the remark this answers is not in the document]
Comment (Ubisoft-potato) [reply pkg/batch/cmd/server/server.go]: Done!

File: pkg/batch/jobs/notification/domain_event_informer.go [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+		logger:            options.logger.Named("sender"),
+	}
+}
+
+func (i *domainEventInformer) Run(ctx context.Context) error {
+	i.logger.Info("DomainEventInformer start running")
+	runningDurationCtx, cancel := context.WithTimeout(ctx, i.opts.runningDurationPerBatch)
+	defer cancel()
+	i.group.Go(func() error {
+		return i.puller.Run(runningDurationCtx)
```
Comment (Ubisoft-potato) [inline pkg/batch/jobs/notification/domain_event_informer.go:127]: @cre8ivejp About if we can remove goroutine in `domainEventInformer`, I found that the documentation of [`pubsub Receive` method](https://pkg.go.dev/cloud.google.com/go/pubsub@v1.31.0#Subscription.Receive):

>Receive calls f with the outstanding messages from the subscription. It **_blocks until ctx is done_**, or the service returns a non-retryable error.

So we can't implement this without goroutine.

**Review by masaaania — APPROVED** [review]
LGTM!!

**Pull request merged**